src/pages/storage/MigrateVolumeBtn.tsx

```diff
@@ -0,0 +1,137 @@
+import { FC, useState } from "react";
+import { ActionButton, Icon } from "@canonical/react-components";
+import usePortal from "react-useportal";
+import { useQueryClient } from "@tanstack/react-query";
+import { queryKeys } from "util/queryKeys";
+import { useEventQueue } from "context/eventQueue";
+import ItemName from "components/ItemName";
+import { useToastNotification } from "context/toastNotificationProvider";
+import { LxdStorageVolume } from "types/storage";
+import MigrateVolumeModal from "./MigrateVolumeModal";
+import { migrateStorageVolume } from "api/storage-pools";
+import { useLocation, useNavigate } from "react-router-dom";
+
+interface Props {
+  storageVolume: LxdStorageVolume;
+  project: string;
+  classname?: string;
+  onClose?: () => void;
+}
+
+const MigrateVolumeBtn: FC<Props> = ({
+  storageVolume,
+  project,
+  classname,
+  onClose,
+}) => {
+  const eventQueue = useEventQueue();
+  const toastNotify = useToastNotification();
+  const { openPortal, closePortal, isOpen, Portal } = usePortal();
+  const queryClient = useQueryClient();
+  const navigate = useNavigate();
+  const [isVolumeLoading, setVolumeLoading] = useState(false);
+  const location = useLocation();
+  const initialURL = location.pathname;
+
+  const handleSuccess = (
+    newTarget: string,
+    storageVolume: LxdStorageVolume,
+    initialURL: string,
+  ) => {
+    toastNotify.success(
+      <>
+        Volume <ItemName item={{ name: storageVolume.name }} bold />{" "}
+        successfully migrated to <ItemName item={{ name: newTarget }} bold />
+      </>,
+    );
+
+    const newStorageVolumeURL = `/ui/project/${storageVolume.project}/storage/pool/${newTarget}/volumes/${storageVolume.type}/${storageVolume.name}`;
+    if (initialURL != newStorageVolumeURL) {
```

Comment:
```suggestion
    const oldVolumeUrl = `/ui/project/${storageVolume.project}/storage/pool/${storageVolume.pool}/volumes/${storageVolume.type}/${storageVolume.name}`;
    const newVolumeUrl = `/ui/project/${storageVolume.project}/storage/pool/${newTarget}/volumes/${storageVolume.type}/${storageVolume.name}`;
    if (startsWith(location.pathname, oldVolumeUrl)) {
```
Two issues need solving:
1. We should compare the current active url, we don't need to pass the initial url around. That would be the url on submit of the request, but we want to know the current page the user is on.
2. We need to check the prefix of the url, because the user might be on the configuration or snapshots subpage with a nested url.

Reply:
Apologies, this has been done within my local branch but I am having issues pushing them. I will fix this and have the old updates and new updates included.